### ISSUE TITLE
cmake not honoring changes to PROJ_BIN_SUBDIR and PROJ_LIB_SUBDIR

### ISSUE THREAD

**Issue by pstieber** [OSGeo/PROJ#817]
Using cmake 3.10.2 on Windows 10 with Visual Studio 2017 version 15.4.4.

Tried to append x64 to bin and local/lib, but this was ignored during installation.

**Comment by kbevers** [comment]
I have put this in for version 5.1 since I don't want to risk cocking something up in the build systems this far into the release process.

My CMake skills are limited. Can you elaborate on what it is you are doing and what you expect to happen. Just so we have a better chance of getting things right.

Your are of course welcome to submit a PR that fixes the problem.

**Comment by pstieber** [comment]
I am using the cmake GUI interface to change the install location by altering `CMAKE_INSTALL_PREFIX`.  This works. I'm also trying to change the binary and library paths by altering `PROJ_BIN_SUBDIR` and `PROJ_LIB_SUBDIR`, but this isn't working. Here's a snapshot of the cmake GUI setup I'm using.  Only the modified `PROJ_BIN_SUBDIR` is listed.  The binaries are installed to `C:/ExternalPackages/bin/` instead of `C:/ExternalPackages/bin/x64/`

![image](https://user-images.githubusercontent.com/5789224/36608659-52838578-187f-11e8-84c4-fa9083613353.png)

**Comment by cffk** [comment]
Running cmake from the command line, e.g.,

    cmake -D CMAKE_INSTALL_PREFIX=/tmp/aa -D PROJ_BIN_SUBDIR=binxx -D PROJ_LIB_SUBDIR=libxx ..

works as expected.  E.g., binaries go into /tmp/aa/binxx.

**Comment by pstieber** [comment]
On a different Windows machine than the one I used for my my original post, I just ran:
```
cmake -G "Visual Studio 15 2017 Win64" -D CMAKE_INSTALL_PREFIX=C:/Junk/proj.4 -D PROJ_BIN_SUBDIR=bin/x64 -D PROJ_LIB_SUBDIR=share/lib/x64 ../proj.4
cmake --build . --config Release --target INSTALL 2>&1 | tee BuildAndInstallLog.txt
```
from the git bash prompt, and the binaries were in `C:/Junk/bin/x64`, but even though the output indicated the library was being installed:
```
  -- Installing: C:/Junk/proj.4/share/lib/x64/proj_5_0.lib
```
it was not there.
I'm using:
Visual Studio 2017 Community Edition version 15.5.7
cmake 3.10.2
Git for Windows 2.16.2.windows.1
master branch at a5f4f52d0050ba5e171962d0640797a76e33aca7

**Comment by cffk** [comment]
I can't reproduce your problem.  I'm using cmake 3.10, Visual Studio 2017 Professional, and the
same commands as you.  `C:/Junk/proj.4/share/lib/x64/proj_5_0.lib` is created.

**Comment by pstieber** [comment]
Thanks for trying and your patience.  I get different results on different machines. I'm hoping you can help me figure out what I'm doing wrong or what versions of cmake, VS, git... are causing the problem.

I went back to the original machine that has:
Visual Studio 2017 Professional Edition Version 15.4.4
cmake 3.10.2
Git for Windows 2.12.2.windows.1
master branch at a5f4f52d0050ba5e171962d0640797a76e33aca7

I ran the following command.  I cut and paste the commands to make sure my originals did not contain typos:
```
cmake -G "Visual Studio 15 2017 Win64" -D CMAKE_INSTALL_PREFIX=C:/Junk/proj.4 -D PROJ_BIN_SUBDIR=bin/x64 -D PROJ_LIB_SUBDIR=share/lib/x64 ../proj.4 2>&1 | tee CmakeLog.txt
cmake --build . --config Release --target INSTALL 2>&1 | tee BuildAndInstallLog.txt
```
and have attached the created log files.  This version's logs do not even attempt to install the library. Would you mind doing the same so I can figure out what's different?
[CmakeLog.txt](https://github.com/OSGeo/proj.4/files/1756698/CmakeLog.txt)
[BuildAndInstallLog.txt](https://github.com/OSGeo/proj.4/files/1756699/BuildAndInstallLog.txt)

**Comment by pstieber** [comment]
Ugh!  I entered `share/lib/x64` instead of `local/lib/x64` and was looking in the wrong place for the file.  The command line version works.  Sorry for the noise.  If the cli works, the GUI version probably does too.

**Comment by cffk** [comment]
Not to worry.  I'm glad the problem is resolved.

**Comment by pstieber** [comment]
I found the solution to my original problem.  In addition to the `PROJ_BIN_SUBDIR` and `PROJ_LIB_SUBDIR` entries, there are also "Ungrouped entries" called `BINDIR` and `LIBDIR` (see attached screen shot.  These seem to overrule the `PROJ_*` versions.  If I changed `BINDIR` and `LIBDIR`, the result worked! Why the GUI version of cmake differs from the CLI, I don't know.

![image](https://user-images.githubusercontent.com/5789224/36651408-ad347d4e-1a5d-11e8-92dc-fa74c5b4c8b5.png)
